src/actions/noting/reset-note-barriers.js

```diff
@@ -17,32 +18,43 @@ module.exports = function resetNoteBarriers(focus, tagName = config.get('default
     errorHandle('Only a valid VFocus element can be passed to resetNoteBarriers, you passed: ', focus);
   }
 
-  //remove any note barriers that exist
-  focus.filter(isVText).forEach(focus => {
-    focus.vNode.text = focus.vNode.text.replace(/\u200B/g, '');
-  });
-
   //add new note barriers
   findAllNotes(focus, tagName).forEach(noteSegments => {
-    //first note
-    noteSegments[0].next().insertBefore(createNoteBarrier());
 
 
-    //last note
-    // This is necessarily complex (been through a few iterations) because
-    // of Chrome's lack of flexibility when it comes to placing the caret.
-    var lastNote = noteSegments.slice(-1)[0].find((node)=> isNotWithinNote(node, tagName));
-    //find the first non-empty text node after the note
-    var adjacentTextNode = lastNote && lastNote.find(isNotEmpty);
-    var adjacentTextNodeContainsOnlyLink = adjacentTextNode && adjacentTextNode.vNode.text && !!adjacentTextNode.vNode.text.match(/^https?:\/\//);
+    //remove any note barriers that exist
+    noteSegments.forEach((note)=>{
+      note.filter(isVText).forEach(focus => {
+        focus.vNode.text = focus.vNode.text.replace(/\u200B/g, '');
+      });
+    });
+
+
+    //add zero width space to first note segment first note
+    noteSegments[0].next().insertBefore(createNoteBarrier());
 
-    if(adjacentTextNodeContainsOnlyLink){
-      adjacentTextNode.parent.addChild(new VText('\u200B'))
-    } else if (adjacentTextNode){
-      adjacentTextNode.vNode.text = '\u200B' + adjacentTextNode.vNode.text;
+    //insert a note barrier after the current note
+    var endingNoteSegment = noteSegments.slice(-1)[0];
+    var nextNode = endingNoteSegment.find((node)=> isNotWithinNote(node, tagName));
+
+    //check whether the adjacent node is a child of the notes parent
+    //if not the note is at the end of a paragraph and the caret needs to be placed within that paragraph
+    //NOT within the adjacent node
+    var isWithinSameElement = !!nextNode
+      ? (endingNoteSegment.parent.vNode.children.indexOf(nextNode.vNode) !== -1)
+      : false;
+
+    if (!isWithinSameElement) {
+      endingNoteSegment.parent.addChild(new VText('\u200B'));
+    } else {
+      var index = nextNode.parent.vNode.children.indexOf(nextNode.vNode);
```

Comment:
Not a blocker but is there anything we can do about the long accesses here? Would be nice to encapsulate it somewhere.